### PR TITLE
Describe how OSCORE works with non-traditional responses

### PR DESCRIPTION
**Pull request by chrysn**
I'm tempted to CC Göran and Marco to this, but in the interest of the timeline I'd hope we can upload this, and once uploaded ask the OSCORE authors for a review.

### CODE REVIEW

File: draft-bormann-core-responses.md [hunk 1]
```diff
@@ -161,6 +161,73 @@ generated by the client, or in close collaboration with the client (for
...
+
+  <!-- Conveniently, this is obsoleting some text that's rotting away in lwig-oscore. -->
+
+  As a convenient effect, this generalized rule also implies
+  that when a server performs {{Appendix B.1.2 of oscore}},
+  it needs to use its own Partial IV for the nonce
+  (which without this generalized rule necessitated a "MUST" statement in the appendix).
+
+[^maybealwaysfirst]: CA: We could also just mandate the "either the first or never" behavior.
+  I don't know why one would save the efficient response,
```
Comment (cabo) [inline draft-bormann-core-responses.md]: What does "save the efficient response" mean here?

File: draft-bormann-core-responses.md [hunk 2]
```diff
@@ -161,6 +161,73 @@ generated by the client, or in close collaboration with the client (for
...
+  but that may be lack of imagination.
+  An affine type system can make this doable in a safe way.
```
Comment (cabo) [inline draft-bormann-core-responses.md]: What does "affine type system" mean here?

File: draft-bormann-core-responses.md (hunk 1 above)
Comment (chrysn) [reply draft-bormann-core-responses.md]: Clarified: The one where the OSCORE option is empty.

File: draft-bormann-core-responses.md (hunk 2 above)
Comment (chrysn) [reply draft-bormann-core-responses.md]: Made explicit: Where items are used at most once, and not generally duplicated.

Comment (cabo) [reply draft-bormann-core-responses.md]: I pushed the scary word into a parentheses.

**Pull request merged**